src/ui/events-list-view.cpp

```diff
@@ -100,16 +101,32 @@ void EventItemDelegate::paint(QPainter *painter,
         avatar = AvatarService::instance()->getAvatar(event.author);
     }
     if (avatar.size() != QSize(kAvatarWidth, kAvatarHeight)) {
-        avatar = QImage(":/images/account-36.png");
+        avatar = QImage(":/images/account.png");
     }
+
+    QRect actualRect(0, 0, kAvatarWidth * devicePixelRatio() , kAvatarHeight * devicePixelRatio());
+    avatar.scaled(actualRect.size());
+    QImage masked_image(actualRect.size(), QImage::Format_ARGB32_Premultiplied);
+    masked_image.fill(Qt::transparent);
+    QPainter mask_painter;
+    mask_painter.begin(&masked_image);
+    mask_painter.setRenderHint(QPainter::Antialiasing);
+    mask_painter.setCompositionMode(QPainter::CompositionMode_SourceOver);
+    mask_painter.setBrush(Qt::white);
+    mask_painter.drawEllipse(actualRect);
+    mask_painter.setCompositionMode(QPainter::CompositionMode_SourceIn);
+    mask_painter.drawImage(actualRect, avatar);
+    mask_painter.setCompositionMode(QPainter::CompositionMode_DestinationOver);
+    mask_painter.fillRect(actualRect, Qt::transparent);
+    mask_painter.end();
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 0, 0))
```

Comment:
The code between line 109 to 121 is the same as those in line 214 to 226 of account-view.cpp except for one argument (`image` v.s. `avatar`). It's better to refactor them out to make it more maintainbale, and reusable latter.

```cpp
// events-list-view.cpp
QImage masked_image(actualRect.size(), QImage::Format_ARGB32_Premultiplied);
masked_image.fill(Qt::transparent);
QPainter mask_painter;
mask_painter.begin(&masked_image);
mask_painter.setRenderHint(QPainter::Antialiasing);
mask_painter.setCompositionMode(QPainter::CompositionMode_SourceOver);
mask_painter.setBrush(Qt::white);
mask_painter.drawEllipse(actualRect);
mask_painter.setCompositionMode(QPainter::CompositionMode_SourceIn);
mask_painter.drawImage(actualRect, avatar);
mask_painter.setCompositionMode(QPainter::CompositionMode_DestinationOver);
mask_painter.fillRect(actualRect, Qt::transparent);
mask_painter.end();
```
```cpp
// account-view.cpp
QImage masked_image(actualRect.size(), QImage::Format_ARGB32_Premultiplied);
masked_image.fill(Qt::transparent);
QPainter mask_painter;
mask_painter.begin(&masked_image);
mask_painter.setRenderHint(QPainter::Antialiasing);
mask_painter.setCompositionMode(QPainter::CompositionMode_SourceOver);
mask_painter.setBrush(Qt::white);
mask_painter.drawEllipse(actualRect);
mask_painter.setCompositionMode(QPainter::CompositionMode_SourceIn);
mask_painter.drawPixmap(actualRect, image);
mask_painter.setCompositionMode(QPainter::CompositionMode_DestinationOver);
mask_painter.fillRect(actualRect, Qt::transparent);
mask_painter.end();
```